### ISSUE TITLE
Forwarding signals

### ISSUE THREAD

**Issue by aeisenbarth** [pyapp-kit/psygnal#70]
A use case that intuitively seems reasonable to me is when you have three objects `A`, `B`, `C` where `A` knows `B` (but not `C`) and `B` knows `C`:
```
A → B
B → C
```

```python
from psygnal import Signal

class A:
    signal1 = Signal()
    
class B:
    signal2 = Signal()
    
a = A()
b = B()
a.signal1.connect(b.signal2.emit)
# and later b.signal2.connect(c.some_method) …
```
This gives the error:
```python
TypeError: cannot create weak reference to 'SignalInstance' object
```

Shouldn't it work, or is the approach wrong?

It works when I don't connect the bound method `emit` directly but wrap it into another function/lambda: 
```python
a.signal1.connect(lambda: b.signal2.emit())
```

**Comment by tlambert03** [comment]
Ah!  Good one.  Yes this is probably because I've defined `__slots__` for performance.  I need to make them weak referenceable,  thanks!

**Comment by aeisenbarth** [comment]
Thanks! 

I just found it myself. But if it's for performance, I'm not sure whether my example is just an edge case and really worth it. But in case someone else hits this, then the above at least documents the workaround.

[Notes on using __slots__](https://docs.python.org/3/reference/datamodel.html#notes-on-using-slots)

> Without a __weakref__ variable for each instance, classes defining __slots__ do not support weak references to its instances. If weak reference support is needed, then add '__weakref__' to the sequence of strings in the __slots__ declaration.

**Comment by tlambert03** [comment]
fix in #71 ... thanks @aeisenbarth
